tests/tck-build-logic/src/main/java/org/graalvm/internal/tck/ContributionTask.java

```diff
@@ -0,0 +1,586 @@
+        sb.indent();
+        sb.openObject().appendStringProperty("excludeClasses", "**").closeObject().concat().newLine();
+        for (int i = 0; i < packages.size(); i++) {
+            String nextPackage = packages.get(i) + ".**";
+            sb.openObject().appendStringProperty("includeClasses", nextPackage).closeObject();
+            if (i < packages.size() - 1) {
+                sb.concat();
+            }
+
+            sb.newLine();
+        }
+
+        sb.unindent();
+        sb.closeArray().newLine();
+        sb.unindent();
+        sb.closeObject();
+
+        writeToFile(testsDirectory.resolve(USER_CODE_FILTER_FILE), sb.toString(), StandardOpenOption.CREATE);
+    }
+
+    private void addAdditionalDependencies(List<Coordinates> dependencies) {
+        if (dependencies == null) {
+            return;
+        }
+
+        Path buildFilePath = testsDirectory.resolve(BUILD_FILE);
+        InteractiveTaskUtils.printUserInfo("Adding following dependencies to " + BUILD_FILE + " file: " + dependencies);
+
+        if (!Files.exists(buildFilePath) || !Files.isRegularFile(buildFilePath)) {
+            throw new RuntimeException("Cannot add additional dependencies to " + buildFilePath + ". Please check if a " + BUILD_FILE + " exists on that location.");
+        }
+
+        try {
+            ConfigurationStringBuilder sb = new ConfigurationStringBuilder();
+            List<String> content = Files.readAllLines(buildFilePath);
+            for (var line : content) {
+                sb.append(line).newLine();
+                if (line.trim().equalsIgnoreCase("dependencies {")) {
+                    sb.indent();
+                    for (var dependency : dependencies) {
+                        sb.append("testImplementation").space().quote(dependency.toString()).newLine();
+                    }
+                    sb.unindent();
+                }
+            }
+
+            writeToFile(buildFilePath, sb.toString(), StandardOpenOption.WRITE);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private void addAgentConfigBlock() {
+        Path buildFilePath = testsDirectory.resolve(BUILD_FILE);
+        InteractiveTaskUtils.printUserInfo("Configuring agent block in: " + BUILD_FILE);
+
+        if (!Files.exists(buildFilePath) || !Files.isRegularFile(buildFilePath)) {
+            throw new RuntimeException("Cannot add additional dependencies to " + buildFilePath + ". Please check if a " + BUILD_FILE + " exists on that location.");
+        }
+
+
+        try(InputStream stream = ContributionTask.class.getResourceAsStream("/contributing/agent.template")) {
+            if (stream == null) {
+                throw new RuntimeException("Cannot find template for the graalvm configuration block");
+            }
+
+            String content = "\n" + (new String(stream.readAllBytes(), StandardCharsets.UTF_8));
```

Comment:
"\n" under Windows will lead to some problems, I guess.

Reply:
I replaced `\n` with `System.lineSeparator()`